test/functional/azure_rm_test.rb

```diff
@@ -4,19 +4,18 @@ module ForemanAzureRM
   class ComputeResourcesControllerTest < ActionController::TestCase
     tests ::ComputeResourcesController
 
-    setup { Fog.mock! }
-    teardown { Fog.unmock! }
-
     test "should create a compute resource and return edit page" do
-      test_client = Fog::Resources::AzureRM.new(
+      test_sdk = ForemanAzureRM::AzureSDKAdapter.new(
```

Comment:
@ShimShtein Do you think this test_sdk object should be defined outside the test so that other tests would use it too? Can we do:
```
test_sdk = ForemanAzureRM::AzureSDKAdapter.new(
        tenant_id: '',
        client_id: '',
        client_secret: '',
        subscription_id: '')
test "should create a compute resource and return edit page" do
.......
end

```

Reply:
You can make the sdk mock available in all tests through a class variable in `setup` block:

``` ruby
class ComputeResourcesControllerTest
  setup do
      @test_sdk = mock('test_sdk') #creates an empty object
      ForemanAzureRM::AzureRM.any_instance.stubs(:sdk).returns(test_sdk)
      @test_resource_client = mock('resource_client')
      test_sdk.stubs(:resource_client).returns(test_resource_client)
  end

  test 'do some tests'
    @test_resource_client.stubs(:rgs).returns([])
     # ...
  end

  test 'do other tests'
    @test_resource_client.stubs(:rgs).returns(['a', 'b', 'c'])
     # ...
  end
end